src/cropduster.js

```diff
@@ -238,17 +260,24 @@ const CD = {
     options = args[1] || {};
     var msg = "getImage: " + url;
 
-    var img = new Image();
-    img.onload = function() {
-      CD.capture(msg);
-      if(callback) { callback(img); }
-    };
-    img.onerror = function() {
-      CD.capture(msg);
-      callback(null);
-    };
-    img.src = url;
-    CD.suspend(options.maxSuspension, msg);
+
+    return new Promise(function(resolve, reject) {
```

Comment:
I guess the same feedback applies to `getImage` and `getImages` too...can there be separate methods for the promise-based implementations? 

It also looks like we do some complex argument stuff so there's the optional `options` argument — but it looks we only use `maxSuspension` on it. We could change the `getImagePromise` signature to be more explicit:

Something like:

```js
getImage(...args) {
  callback = args.pop();
  url = args[0];
  options = args[1] || {};

  this.getImagePromise(url, options.maxSuspension).then(
    img => callback(img),
    _ => callback(null)
  );
}
```